runtime/oti/vm_api.h

```diff
@@ -554,10 +554,11 @@ internalExceptionDescribe(J9VMThread *vmThread);
 * @param lineNumber)
 * @param userData
 * @param pruneConstructors
+* @param nbrCacheItems
 * @return UDATA
 */
 UDATA
-iterateStackTrace(J9VMThread * vmThread, j9object_t* exception,  UDATA  (*callback) (J9VMThread * vmThread, void * userData, J9ROMClass * romClass, J9ROMMethod * romMethod, J9UTF8 * fileName, UDATA lineNumber, J9ClassLoader* classLoader), void * userData, UDATA pruneConstructors);
+iterateStackTrace(J9VMThread * vmThread, j9object_t* exception,  UDATA  (*callback) (J9VMThread * vmThread, void * userData, U_8 *methodPC, J9ConstantPool *constantPool, J9ROMClass * romClass, J9ROMMethod * romMethod, J9UTF8 * fileName, UDATA lineNumber, J9ClassLoader* classLoader), void * userData, UDATA pruneConstructors, U_32 nbrCacheItems);
```

Comment:
Fixed.

@DanHeidinga please have another look.

Reply:
Just realized that `iterateStackTrace()` can't depend on the incoming `numCacheItems` to find the cached `constantPool`, otherwise there is a possibility that some `iterateStackTrace()` calls might not correctly iterate the `walkback` when `NPE` extended message is enabled with additional J9_STACKWALK_CACHE_CPS.

Added a helper method `isNPEMsgEnabled()` to determine if the `constantPool` entry presents or not.

Please review.